port/raspberrypi/rp2xxx/src/hal/pio/common.zig

```diff
@@ -225,15 +243,24 @@ pub fn PioImpl(EnumType: type, chip: Chip) type {
             });
         }
 
-        pub fn sm_set_exec_options(self: EnumType, sm: StateMachine, options: ExecOptions) void {
+        fn pin_to_index(self: EnumType, pin: gpio.Pin) error{InvalidPin}!u5 {
+            const index = @intFromEnum(pin);
+            const base = (0x10 & self.get_regs().GPIOBASE.raw);
+            if (index < base or index >= base + 32) {
```

Comment:
Is this OK for rp2350? the 32 is the number of pins that be setup in pio, not how many pins the chip has, correct?